### PR TITLE
[Authz] extract catalog for ResolvedTable and DataSourceV2Relation

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

extract catalog for ResolvedTable and DataSourceV2Relation in tableExtractors

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @yaooqinn

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala [hunk 1]
```diff
@@ -89,4 +89,56 @@ class V2JdbcTableCatalogPrivilegesBuilderSuite extends V2CommandsPrivilegesSuite
...
+
+  test("Extracting table info with ResolvedTableTableExtractor") {
+    val ns1 = "testns1"
+    val tbl = "testtbl"
+    withDatabase(s"$ns1") { ns =>
+      sql(s"CREATE NAMESPACE $ns")
+      withTable(s"$catalogV2.$ns1.$tbl") { t =>
+        sql(s"CREATE TABLE IF NOT EXISTS $t(key int)")
+        Seq(
+          s"SHOW CREATE TABLE $t").foreach { str =>
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala:101]: Are you planning more test cases here?

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala [hunk 2]
```diff
@@ -89,4 +89,56 @@ class V2JdbcTableCatalogPrivilegesBuilderSuite extends V2CommandsPrivilegesSuite
...
+
+  test("Extracting table info with DataSourceV2RelationTableExtractor") {
+    val ns1 = "testns1"
+    val tbl = "testtbl"
+    withDatabase(s"$ns1") { ns =>
+      sql(s"CREATE NAMESPACE $ns")
+      withTable(s"$catalogV2.$ns1.$tbl") { t =>
+        sql(s"CREATE TABLE IF NOT EXISTS $t(key int)")
+        Seq(
+          s"INSERT INTO TABLE $t VALUES(1)").foreach { str =>
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala:127]: ditto

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala (hunk 1 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala:101]: It is good enough to cover all 3 table extractors for datasource v2 logical plans, including `ResolvedTable` and `DataSourceV2Relation` in this PR. For the rest in v1, the `catalog `of `Table` remains to be None by default.

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/V2JdbcTableCatalogPrivilegesBuilderSuite.scala:101]: Thus, let's avoid his seq?

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**